scipy/interpolate/rbf.py

```diff
@@ -200,11 +200,13 @@ def __init__(self, *args, **kwargs):
         r = self._call_norm(self.xi, self.xi)
         self.epsilon = kwargs.pop('epsilon', None)
         if self.epsilon is None:
-            # default epsilon is the "the average distance between nodes"
+            # default epsilon is the "the average distance between nodes" based
+            # on a bounding hypercube
             dim = self.xi.shape[0]
             ximax = np.amax(self.xi, axis=1)
             ximin = np.amin(self.xi, axis=1)
-            self.epsilon = np.power(np.prod(ximax - ximin)/self.N, 1.0/dim)
+            edges = np.array([a for a in ximax-ximin if a != 0.0])
```

Comment:
this is probably better written as e.g. `edges = ximax - ximin;  edges = edges[np.nonzero(edges)]`.